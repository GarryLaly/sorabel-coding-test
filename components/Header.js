import Head from "next/head";
import React from "react";
import PropTypes from "prop-types";
import Link from "next/link";

const Header = ({ title }) => (
  <div>
    <Head>
      <meta charSet="utf-8" />
      <meta
        name="viewport"
        content="width=device-width, initial-scale=1, shrink-to-fit=no"
      />
      <link
        rel="stylesheet"
        href="../static/vendor/bootstrap/css/bootstrap.min.css"
      />
      <link rel="stylesheet" href="../static/css/style.css" />
      <link
        href="https://fonts.googleapis.com/css?family=Poppins:400,500,600"
        rel="stylesheet"
      />
      <title>{title}</title>
      <script type="text/javascript" src="../static/js/jquery.min.js" />
      <script
        type="text/javascript"
        src="../static/vendor/bootstrap/js/bootstrap.min.js"
      />
      <script type="text/javascript" src="../static/js/popper.min.js" />
    </Head>
    <header>
      <div className="navbar navbar-dark bg-dark shadow-sm">
        <div className="container d-flex justify-content-between">
          <Link href="/">
            <a href="/" className="navbar-brand d-flex align-items-center">
              <strong>{title}</strong>
            </a>
          </Link>
        </div>
      </div>
    </header>
  </div>
);

Header.propTypes = {
  title: PropTypes.string.isRequired
};

export default Header;
