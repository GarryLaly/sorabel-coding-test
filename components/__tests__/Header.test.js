import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Header from "../Header";

describe("Header", () => {
  const setup = propOverrides => {
    const props = {
      title: "Web Title",
      ...propOverrides
    };

    const wrapper = shallow(<Header {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
