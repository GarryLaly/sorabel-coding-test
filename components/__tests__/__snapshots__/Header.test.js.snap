// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders default 1`] = `
<div>
  <Head>
    <meta
      charSet="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
      name="viewport"
    />
    <link
      href="../static/vendor/bootstrap/css/bootstrap.min.css"
      rel="stylesheet"
    />
    <link
      href="../static/css/style.css"
      rel="stylesheet"
    />
    <link
      href="https://fonts.googleapis.com/css?family=Poppins:400,500,600"
      rel="stylesheet"
    />
    <title>
      Web Title
    </title>
    <script
      src="../static/js/jquery.min.js"
      type="text/javascript"
    />
    <script
      src="../static/vendor/bootstrap/js/bootstrap.min.js"
      type="text/javascript"
    />
    <script
      src="../static/js/popper.min.js"
      type="text/javascript"
    />
  </Head>
  <header>
    <div
      className="navbar navbar-dark bg-dark shadow-sm"
    >
      <div
        className="container d-flex justify-content-between"
      >
        <Link
          href="/"
        >
          <a
            className="navbar-brand d-flex align-items-center"
            href="/"
          >
            <strong>
              Web Title
            </strong>
          </a>
        </Link>
      </div>
    </div>
  </header>
</div>
`;
